lib/ansible/module_utils/manageiq.py

```diff
@@ -67,6 +70,36 @@ def validate_connection_params(module):
             module.fail_json(msg=error_str.format(arg))
 
 
+def manageiq_extra_vars(module_vars, task_vars):
+    if 'manageiq' not in task_vars.keys():
+        return module_vars
+
+    verify_ssl = None
+
+    if 'manageiq_connection' not in module_vars.keys() or module_vars['manageiq_connection'] is None:
+        module_vars['manageiq_connection'] = dict()
+    if 'verify_ssl' in module_vars['manageiq_connection'].keys():
+        verify_ssl = module_vars['manageiq_connection'].pop('verify_ssl', None)
+
+    for k in MANAGEIQ_MODULE_VARS:
+        if k not in module_vars['manageiq_connection']:
+            try:
+                module_vars['manageiq_connection'][k] = task_vars['manageiq'][MANAGEIQ_ARGS_MAP[k]]
+            except KeyError:
+                pass
+
+    if verify_ssl:
+        module_vars['manageiq_connection']['verify_ssl'] = verify_ssl
+
+    try:
+        if module_vars['manageiq_connection']['username'] and module_vars['manageiq_connection']['password']:
+            module_vars['manageiq_connection'].pop('token')
```

Comment:
Does this work — is it 'token' or 'api_token' at this point?

Why is this needed?
I guess it's so a playbook specifying user&pwd wins over ansible var specifying token?
IINM, doing this after the merge it can have other effects eg. ansible var specifying user&pwd wins over playbook specifying a token...

* Idea: Do we have to merge at all?  What if we simplify the behavior to "whole `manageiq_connection` param, if unset, defaults to `manageiq_connection` ansible var"?  If someone really needs to specify some params and inherit others from a var, they can do it themselves with {{...}} templating, with full control.